azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java

```diff
@@ -125,8 +125,11 @@ public void setTelemetryProcessors(
 
     @Bean
     @DependsOn("internalLogger")
-    public TelemetryConfiguration telemetryConfiguration(TelemetryChannel telemetryChannel) {
+    public TelemetryConfiguration telemetryConfiguration() {
         TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.getActiveWithoutInitializingConfig();
+        if (StringUtils.isNotEmpty(applicationInsightsProperties.getConnectionString())) {
+            telemetryConfiguration.setConnectionString(applicationInsightsProperties.getConnectionString());
+        }
         telemetryConfiguration.setTrackingIsDisabled(!applicationInsightsProperties.isEnabled());
         telemetryConfiguration.setInstrumentationKey(applicationInsightsProperties.getInstrumentationKey());
```

Comment:
will this throw exception if using connection string and not instrumentation key directly?